### PR TITLE
Fix entity list update after mutation | Closes #2750

### LINKED ISSUE #2750: Tickets entity list does not reflect the change when creating a datetime and a ticket.
Steps:
- Reload EDTR
- Set filters to show all Dates/Tickets and disable `isChained` filter
- Create a datetime
- Now create a Ticket

Expected: Ticket should show up in the list

Actual: Ticket doesn't show up.

Reason: The ticket query options get bound inside ticket mutation handler.

Solution: We may need to extract the update function to a hook.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR fixes the issue caused due to deps not being added to `useCallback`

### CODE REVIEW

File: assets/src/domain/eventEditor/services/apollo/mutations/prices/usePriceMutationHandler.ts [hunk 1]
```diff
@@ -31,42 +32,41 @@ const usePriceMutationHandler = (): MutationHandler => {
...
+						onDeletePrice({ proxy, prices, price, deletePermanently: input?.deletePermanently });
+						break;
+				}
+			};
 
-		return { variables, optimisticResponse, onUpdate };
-	};
+			return { variables, onUpdate };
+		},
+		[options.variables]
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/prices/usePriceMutationHandler.ts:68]: shouldn't `variables` also be included here?

File: assets/src/domain/eventEditor/services/apollo/mutations/useMutationOptions.ts [hunk 2]
```diff
@@ -16,62 +16,65 @@ import { useMutationHandler, mutations } from './';
...
-			...options,
-		};
-	}, []);
+			const options = getMutationOptions();
+			return {
+				mutation: getMutation(),
+				...options,
+			};
+		},
+		[getMutationHandler]
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/useMutationOptions.ts:76]: `typeName` also needs to be included:

image

File: assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useDatetimeMutationHandler.ts [hunk 3]
```diff
@@ -22,36 +23,39 @@ const useDatetimeMutationHandler = (): MutationHandler => {
...
+						break;
+					case MutationType.Delete:
+						onDeleteDatetime({ proxy, datetimes, datetime, deletePermanently: input?.deletePermanently });
+						break;
+				}
+			};
+
+			return { variables, optimisticResponse, onUpdate };
+		},
+		[options.variables]
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/datetimes/useDatetimeMutationHandler.ts:57]: image

File: assets/src/domain/eventEditor/services/apollo/mutations/prices/usePriceMutationHandler.ts (hunk 1 above)
Comment (nicolad) [reply assets/src/domain/eventEditor/services/apollo/mutations/prices/usePriceMutationHandler.ts:68]: image

File: assets/src/domain/eventEditor/services/apollo/mutations/tickets/useTicketMutationHandler.ts [hunk 4]
```diff
@@ -23,42 +24,45 @@ const useTicketMutationHandler = (): MutationHandler => {
...
+						onDeleteTicket({ proxy, tickets, ticket, deletePermanently: input?.deletePermanently });
+						break;
+				}
+			};
 
-		return { variables, optimisticResponse, onUpdate };
-	};
+			return { variables, optimisticResponse, onUpdate };
+		},
+		[options.variables]
```
Comment (nicolad) [inline assets/src/domain/eventEditor/services/apollo/mutations/tickets/useTicketMutationHandler.ts:64]: image

File: assets/src/domain/eventEditor/services/apollo/mutations/useMutationOptions.ts (hunk 2 above)
Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/services/apollo/mutations/useMutationOptions.ts]: Updated

**Pull request merged**